### ISSUE TITLE
reuse more from the EC core ontologies

### ISSUE THREAD

**Issue by VladimirAlexiev** [OP-TED/ePO#295]
AFAIK, ePO reuses only org:Site (and has some common superclasses like foaf:Agent etc).
Should it not reuse more of the "core ontologies" as defined by SEMIC and then adopted by W3C?

In particular:
- org:Organization
- rov:RegisteredOrganization
- locn:Address

**Comment by giorgialodi** [comment]
org:Site has been recently removed but in the new version of the ontology, still not published. We will definitely align the epo ontology to EU core vocabularies and W3C Web standards; please note that we will also revise the location part of the epo ontology, according to the recent revisions of EU vocs.

**Comment by VladimirAlexiev** [comment]
> org:Site has been recently removed ... will definitely align

Will ePO use EU Core vocabs, or just align to them using owl:equivalentClass and the like?

**Comment by giorgialodi** [comment]
This is something that has to be still officially decided in the working group even if some discussions about that have been already faced. I can provide you with my personal opinion that I am bringing to the working group: since the EU vocs are semantic assets created in the context of the same organisation (EU commission) who is guiding here, I do not see any issue to directly reuse those vocabularies when possible, thus creating a sort of network of EU commission's ontologies. For instance, if the location modelling of the epo ontology (which is something quite independent of the procurement domain) is the one that is already available in the Core Location Vocabulary, I would suggest the working group to use directly Core Location Voc. In some cases, there are additional properties to be added for some concepts. In that case, I would suggest to indirectly reuse existing vocs, therefore a separate alignment is better: we define our semantics and then we align it to the Core Vocs. An example can be the CCCEV - Core Criterion and Evidence Vocabulary, which is quite useful for the procurement domain. However, it seems to me that here we define additional elements, so an alignment can be better. 
For any other external ontologies (e.g., W3C) I would suggest an alignment.
My2cents

**Comment by VladimirAlexiev** [comment]
I think it's always better to reuse: this increases the chances of DATA reuse across domains.
Eg the euBusinesssGraph project made EU company data in Org, RegOrg, and additions in an `ebg:` ontology.
If someone wants to integrate that data with procurement data, it'll be easier if procurement data uses

I disagree that it matters so much WHO made the ontology. Furthermore, W3C took the SEMIC core vocab conceptual models and implemented them as ontologies, reusing more of what exists and simplifying in the process. And these are now the official versions of the ontologies. So it's not "EU vs W3C", it is "EU **and** W3C". You can see this here: https://www.w3.org/ns/locn.

If the Core Vocabs are not capable enough to be used in ePO, they should be revised.
But I don't think that's the case:
looking at https://www.w3.org/ns/locn, locn:Adress has 11 props whereas epo:Address has only 4.
This looks like regress to me, and ePO better be able to explain why it's not using locn:Address.

Often a new ontology needs a few new props even when it reuses a class and all its props.
Then we have various courses of action:
1. Talk to the keepers of the original ontology (W3C and SEMIC) to extend it
2. Use the class as is but extend it by using restrictions and/or shapes
3. Define a new subclass and add the new props there

**Comment by giorgialodi** [comment]
Sorry @VladimirAlexiev I never said "EU Vocs vs W3C". Where do you conclude something like this? For some EU Core Vocs there is the corresponding in W3C for specific reasons, but not for all the EU Vocs (e.g., CCCEV, CPSV, CPV). Probably only the CLV has the URI that is "W3C-based". The others not. In additions, some of the EU Core Vocs uses W3C web standards (e.g., CPOV). Additionally,  I am not saying not to reuse. What I am bringing to the discussion is the **type of reuse**. If you align and you materialize in the data the alignments where is the problem?

I do not agree that we need to revise the EU Core Vocs for the epo; if epo needs something more, it extends it in the specific domain it treats (your last solution 3.). EU Core Vocs are designed having in mind specific requirements, one of them is the possibility to be highly extensible by Member States and others and are independent of the specific domain where they are used.
The epo:Address is the following one and does not have only 4 props:

<img width="326" alt="Screenshot 2021-07-19 at 12 01 54" src="https://user-images.githubusercontent.com/24624577/126143174-84f3c8be-6ddc-494d-98bf-c139f6f01836.png">

As I said, from my point of view, for the Location part we may consider to **directly reuse locn** (even if we need to pay attention to the INSPIRE Address data model as well). For others, also for future sustainability of the ontology, I would suggest to align.

The thing that I really do not understand is that most of the people accept for instance that schema.org redefines everything. also primitive types:  nobody complains for not directly reusing a W3C web standard as DCAT for defining dataset or other ontologies schema.org reproduces inside it.  For all the other ontologies, people no longer accept this policy: for other ontologies the must is to **directly reuse** any external ontology available at the state of the art  :)

**Comment by guascce** [comment]
Our approach has been and still is to align with external foundational ontologies when relevant and we gave all our attention to the EU Core vocabularies. 
Though in order to keep stability we chose to create equivalent classes and when needed to extend them to cover domain specific needs. This is not yet defined as owl resource and we still need to assess whether it is feasible to automate these alignments.

**Comment by VladimirAlexiev** [comment]
>epo:Address has only 4 fields

I'm pretty that's what I saw in 
https://github.com/eprocurementontology/eprocurementontology/blob/2.0.1/v2.0.1/02-Analysis%20and%20design/OWL/ePO_restrictions.rdf
But can't recheck right now because I'm on a phone.

**Comment by eprocurementontology** [comment]
The file you point to shows the different attributes of address.  The file is automatically created from the UML file and does not order the attributes in sequence.  A search of the file for address will show you all the attributes.

**Comment by VladimirAlexiev** [comment]
@eprocurementontology why did you close?

As @giorgialodi said "This is something that has to be still officially decided in the working group".

Even if the WG decides not to reuse and extend, it should provide a semantic mapping, so you can rename the task thusly.

**Comment by eprocurementontology** [comment]
This issue is closed because it is being followed under[github issue 266](https://github.com/eprocurementontology/eprocurementontology/issues/266) Sorry for not being clearer

**Comment by VladimirAlexiev** [comment]
@giorgialodi this is what threw me off:
```ttl
epo:Address  a           owl:Class ;
        rdfs:subClassOf  [ a                   owl:Restriction ;
                           owl:maxCardinality  1 ;
                           owl:onProperty      epo:has
                         ] ;
        rdfs:subClassOf  [ a                   owl:Restriction ;
                           owl:maxCardinality  1 ;
                           owl:onProperty      epo:hasCountryCode
                         ] ;
        rdfs:subClassOf  [ a                   owl:Restriction ;
                           owl:maxCardinality  1 ;
                           owl:onProperty      epo:hasID
                         ] ;
        rdfs:subClassOf  [ a                   owl:Restriction ;
                           owl:maxCardinality  1 ;
                           owl:onProperty      epo:hasCountrySubentityCode
                         ] .
```

But then there are extra props:
```ttl
epo:hasAdditionalStreetName
        rdfs:domain  epo:Address ;
        rdfs:range   rdfs:Literal .
epo:hasBlockName  rdfs:domain  epo:Address ;
        rdfs:range   rdfs:Literal .
...
```

Cheers!

**Comment by VladimirAlexiev** [comment]
@eprocurementontology How about `rov:RegisteredOrganization` and `org:Organization`? These are not covered by #266

**Comment by eprocurementontology** [comment]
Indeed you are correct the issue should still be open.  Any discussion on location is transferred to issue 266 however discussion on the general use of other ontologies can be kept here.

**Comment by costezki** [comment]
After reviewing this thread, all raised issues seem to have been addressed already (ePO 3.0.1). 
ePO has been aligned to Core Location (and a few other vocabularies).
